### PR TITLE
[cmake] Build swift-testing with CMake.

### PR DESCRIPTION
**Pull request by jeffdav**
Add files needed to build `swift-testing` with CMake.

### Motivation:

To bring `swift-testing` to Windows, we need to support a non-SPM path to allow building targets that would otherwise hit the export symbol limit on Windows.  

To use replace `XCTest` with `swift-testing` for The Browser Company's various swift-based projects, of which several build for Windows and exceed the export symbol limit on Windows.  

### Modifications:

No code modifications were necessary.  This simply adds `CMakeLists.txt` files where needed and two `.cmake` files included by them.  The first include is common compiler settings applied to each `swift-testing` target, including `TestingMacros` which is built by recursively invoking CMake, as is [the style of the times](https://github.com/apple/swift-cmake-examples/blob/7a86d39fe88271501431f5f7214edbcc5c1f745b/4_swift_macros/CMakeLists.txt#L26-L34).  The second file defines a function that generates the required runner file with the SPM entry point, and defines a CMake variable consumers of `swift-testing` can use to add the generated source to their test executable target.

### Result:

We can all build with CMake.

### Checklist:

- [X] Code and documentation should follow the style of the [Style Guide](https://github.com/apple/swift-testing/blob/main/Documentation/StyleGuide.md).
- [X] If public symbols are renamed or modified, DocC references should be updated.

### CODE REVIEW

**Comment by jeffdav** [comment]
Mainly just looking for feedback on the general approach here.  Still have to do some work on long paths before it can merge.  And I'll probably make a clean fork, not actually merge this.

**Comment by jeffdav** [comment]
Usage example in `swift-webdriver`: https://github.com/thebrowsercompany/swift-webdriver/compare/main...jeff/swift-testing#diff-a41dfc8fb9233e3978ec6229dd608a5e57ace7b00dd82aefeab761581056f683R1

**Review by compnerd — APPROVED** [review]
I don't see anything unexpected/out of the ordinary here. It is slightly unfortunate what we have to do to pass some of the flags (e.g. `-enable-experimental-feature`), but that is currently something that we cannot do anything without alterations to CMake itself.

File: Sources/TestingMacros/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  Support/CRC32.swift
+  Support/DiagnosticMessage.swift
+  Support/DiagnosticMessage+Diagnosing.swift
+  Support/SourceCodeCapturing.swift
+  Support/SourceLocationGeneration.swift
+  TagMacro.swift
+  TestDeclarationMacro.swift
+  TestingMacrosMain.swift)
+set_target_properties(TestingMacros PROPERTIES
+  Swift_MODULE_NAME TestingMacros
```
[the remark this answers is not in the document]
Comment (jeffdav) [reply Sources/TestingMacros/CMakeLists.txt:54]: Oh, this was a dead chicken that we waved.  Maybe I can remove it now.

**Pull request closed without merging**